anxcloud/resource_object_storage_user.go

```diff
@@ -0,0 +1,190 @@
+package anxcloud
+
+import (
+	"context"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"go.anx.io/go-anxcloud/pkg/apis/common"
+	objectstoragev2 "go.anx.io/go-anxcloud/pkg/apis/objectstorage/v2"
+)
+
+func resourceObjectStorageUser() *schema.Resource {
+	return &schema.Resource{
+		Description:   "This resource allows you to configure Object Storage users.",
+		CreateContext: resourceObjectStorageUserCreate,
+		ReadContext:   resourceObjectStorageUserRead,
+		UpdateContext: resourceObjectStorageUserUpdate,
+		DeleteContext: resourceObjectStorageUserDelete,
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Read:   schema.DefaultTimeout(1 * time.Minute),
+			Update: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+		Schema: mergeSchemas(
+			schemaObjectStorageCommon(),
+			schemaObjectStorageState(),
+			map[string]*schema.Schema{
+				"user_name": {
+					Type:        schema.TypeString,
+					Required:    true,
+					Description: "The name of the user.",
+				},
+				"full_name": {
+					Type:        schema.TypeString,
+					Required:    true,
+					Description: "The full name of the user.",
+				},
+				"enabled": {
+					Type:        schema.TypeBool,
+					Optional:    true,
+					Default:     true,
+					Description: "Indicates whether the user is enabled.",
+				},
+				"backend": {
+					Type:        schema.TypeString,
+					Required:    true,
+					Description: "Identifier of the S3 backend this user belongs to.",
+				},
+				"tenant": {
+					Type:        schema.TypeString,
+					Required:    true,
+					Description: "Identifier of the tenant this user belongs to.",
+				},
```

Comment:
Naming of the fields should be consistent:


```suggestion
				"backend_id": {
					Type:        schema.TypeString,
					Required:    true,
					Description: "Identifier of the S3 backend this user belongs to.",
				},
				"tenant_id": {
					Type:        schema.TypeString,
					Required:    true,
					Description: "Identifier of the tenant this user belongs to.",
				},
```